ts/Core/Chart/Chart.ts

```diff
@@ -434,6 +436,10 @@ class Chart {
             const options = merge(defaultOptions, userOptions), // Do the merge
                 optionsChart = options.chart;
 
+            if (options.plotOptions.pie) {
+                this.mergePieDataLabelPlotOptions(options.plotOptions.pie);
+            }
```

Comment:
This should likely happen in `PieSeries.ts`. This seems to work:

```ts
addEvent(PieSeries, 'afterInit', (e): void => {
    const chart = e.target.chart;
    if (chart.options.plotOptions.pie) {
        PieSeries.mergePieDataLabelPlotOptions(chart.options.plotOptions.pie);
    }
});
```

Possibly the `update` or `setOptions` events can be used for processing when changing options.